src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,281 @@
+path = os.path.join(settings.TEMPLATE_DIR)
+sys.path.append(path)
+import validate
+import random
+import math
+
+def get_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--op", help = "give the operators to be tested", nargs = '+',
+    default = ["+", "-", "/", "*", "%", "^"], choices = ["+", "-", "/", "*", "%", "^"])
+    parser.add_argument("--test-count", help = "give the number of tests to run", nargs = 1, default = [500],
+    type = int)
+    parser.add_argument("--output", help = "give the path to generate files into", default = settings.NUMBER_DIR)
+    parser.add_argument("--operand-count", help = "give the number of maximum operands in an expression",
+    nargs = 1, default = [10], choices = range(2, 11), type = int)
+    parser.add_argument("--seed", help = "give the random seed value", nargs = 1, default = [10000],
+    type = int)
+    if len(sys.argv) == 1:
+        parser.print_help()
+        sys.exit(1)
+
+    script_args = parser.parse_args()
+
+    return script_args
+
+filename = '''binary-ops-{NUMBER}.js'''
+MAX = (1<<53) - 1
+MIN = -1 * MAX
+BITMAX = 9999999
+BITMIN = -9999999
+
+class GenSequence:
+    def __init__(self, options, bit):
+        num_operands = options.operand_count[0]
+        self.num_operands = num_operands
+        self.operators_list = []
+        for i in range(len(options.op)):
+            if (bit == False):
+                if (options.op[i] == "+"):
+                    self.operators_list.append(self.add)
+                if (options.op[i] == "-"):
+                    self.operators_list.append(self.sub)
+                if (options.op[i] == "*"):
+                    self.operators_list.append(self.mul)
+                if (options.op[i] == "/"):
+                    self.operators_list.append(self.div)
+                if (options.op[i] == "%"):
+                    self.operators_list.append(self.mod)
+            else:
+                if (options.op[i] == "^"):
+                    self.operators_list.append(self.bitxor)
+        self.operands = [0] * num_operands
+        self.operators = [0] * (num_operands -1)
+
+        if (bit == False):
+            for i in range(num_operands -1):
+                self.operands[i] = random.randint(MIN, MAX)
+                self.operators[i] = random.randint(0, len(self.operators_list) -1)
+            self.operands[num_operands -1] = random.randint(MIN, MAX)
+
+        else:
+            for i in range(num_operands -1):
+                self.operands[i] = random.randint(BITMIN, BITMAX)
+                self.operators[i] = random.randint(0, len(self.operators_list) -1)
+            self.operands[num_operands -1] = random.randint(BITMIN, BITMAX)
+
+    def add(self, op1, op2):
+        sum = op1 + op2
+        if sum < MIN:
+            op2 = MIN - random.randint(MIN, op1)
+            sum = op1 + op2
+        elif sum > MAX:
+            op2 = MAX - random.randint(op1, MAX)
+            sum = op1 + op2
```

Comment:
Ditto.

Reply:
I need this, because in this case I modify `op2` and so I have to modify `sum`